docs/concepts.md

```diff
@@ -0,0 +1,75 @@
+# ddc concepts
+
+<!--
+Copyright (C) The ddc development team, see COPYRIGHT.md file
+
+SPDX-License-Identifier: MIT
+-->
+
+DDC introduces labels in the form of dimensions and attributes on top of Kokkos arrays, which allows for a more intuitive and less error-prone developer experience.
+
+In fact, in Kokkos, the indices of views are weakly typed, meaning that each index is a simple integer. Let's consider a multidimensional view intended to represent a physical quantity, such as a velocity field for example. The first index represents the velocity along the X-axis, and the second index represents the velocity along the Y-axis. As both are simply integers this can lead to a situation where one mistakenly swaps the two indices. The code would compile successfully, but the resulting behavior would be incorrect, and the source of the error could be difficult to trace.
+
+The advantage of using DDC is that it provides array-like containers that have labeled dimensions using strongly typed indices. For instance, by labeling dimensions `X` and `Y`, the indices along those labeled dimensions become strongly typed preventing the user from making such mistakes.
+
+\remark Note that the use of DDC is not restricted to solving equations. Indeed, one can easily imagine strong typing of variables corresponding to names or ages in a database. The operations must then be adapted accordingly. Here, we largely base our approach on the uniform and non-uniform resolution of the heat equation in two dimensions, which is why we focus the presentation on solving equations using finite differences on a 2D grid.
```

Comment:
By "here" I meant "In this document". Maybe better to make it explicit as you say.

I'm not sure if it's really relevant to mention the 'data' aspect of DDC here; in my opinion, it might confuse the reader. Perhaps we should develop a separate example and dedicate documentation to it ?